libs/@local/graph/store/src/entity_type/store.rs

```diff
@@ -246,6 +249,16 @@ pub struct UpdateEntityTypeEmbeddingParams<'a> {
     pub reset: bool,
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+#[cfg_attr(feature = "utoipa", derive(utoipa::ToSchema))]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct HasPermissionForEntityTypesParams<'a> {
+    #[cfg_attr(feature = "utoipa", schema(value_type = String))]
+    pub action: ActionName,
+    pub entity_type_ids: Cow<'a, [VersionedUrl]>,
+    pub include_parents: bool,
```

Comment:
Can we document how this changes the behavior of the check?

Reply:
We agreed offline that we're going to remove this paramter. Instead, we will determine based on the action what we do. Further, we're going to remove the `instantiate` permission in the future: H-4956